### DISCUSSION TITLE
Possible additional changes to be added to PR #100

### DISCUSSION THREAD

**Discussion by aivarsoo** [PKU-Alignment/safety-gymnasium#101] [Ideas]
Following up on the PR #100. I additionally propose to make the following changes:

1. Replace the classes `Goal0`, `Goal1` and `Goal2` etc (all the navigation tasks) with the config dictionaries. 
2. Perform similar changes in the vision environments relying on the `Builder` class.
3. Store the task name in the `config` dictionary for all tasks in the navigation environments. This will simplify the environment creation process in the [`Builder` class](https://github.com/aivarsoo/safety-gymnasium/blob/main/safety_gymnasium/builder.py#L147), i.e., there will be no need for the `get_task_class_name` method. This will also potentially allow to register just one base environment for customization streamlining the process. The code will be 
```python
env = safety_gymnasium.make('SafetyNavigation-v0', config={'agent_name': ..., 'task_name': ..., }
```

Please let me know if this is aligned with the library development and makes sense to implement

**Comment by muchvo** [comment]
Thank you for your contribution! I'll be meticulously reviewing your work over the next few days and will discuss any points with you as needed.

**Comment by muchvo** [comment]
Your contributions have been successfully merged into the main branch. Thank you for your meticulous observation, outstanding code quality, and comprehensive tests. Reading your code convinced me that you have a deep understanding of our library, and this feature will significantly enhance the user experience for many. Additionally, I apologize for the delayed merge.
